src/client/metadataApiDeploy.ts

```diff
@@ -408,7 +408,11 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
       const zip = createArchive('zip', { zlib: { level: 9 } });
       // anywhere not at the root level is fine
       zip.directory(this.options.mdapiPath, 'zip');
-      await zip.finalize();
```

Comment:
this was changed in here https://github.com/forcedotcom/source-deploy-retrieve/pull/748/files#diff-241374a8e829853ad35d675aec9a8689aa15caa58582f8270c9e4c3de08e3aecL390

the linter/types are unreliable.